python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -259,21 +332,331 @@ def scaleModel(self, model, star, inPlace=True, nb90Rots=0):
+            An annulus of the given star.
+        """
+        # Create SpanSet of annulus
+        outerCircle = SpanSet.fromShape(
+            brightStarStamp.optimalOuterRadius, Stencil.CIRCLE, offset=self.warper.modelCenter
+        )
+        innerCircle = SpanSet.fromShape(
+            brightStarStamp.optimalInnerRadius, Stencil.CIRCLE, offset=self.warper.modelCenter
+        )
+        annulus = outerCircle.intersectNot(innerCircle)
+        return annulus
+
+    def applyStatsControl(self, annulusImage):
+        """Apply statistics control to the PSF annulus image.
+
+        Parameters
+        ----------
+        annulusImage : `~lsst.afw.image.MaskedImageF`
+            An image containing an annulus of the given model.
+
+        Returns
+        -------
+        annularFlux: float
+            The annular flux of the PSF model at the radius where the flux of the given star is determined.
+        """
+        andMask = reduce(
+            ior, (annulusImage.mask.getPlaneBitMask(bm) for bm in self.warper.config.badMaskPlanes)
+        )
+        self.missedStatsControl.setAndMask(andMask)
+        annulusStat = makeStatistics(annulusImage, self.missedStatsFlag, self.missedStatsControl)
+        return annulusStat.getValue()
+
+    def findPsfAnnularFlux(self, brightStarStamp, maskedModel):
+        """Find the annular flux of the PSF model at the radius where the flux of the given star is
+        determined.
+
+        Parameters
+        ----------
+        brightStarStamp : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamp`
+            A stamp of a bright star to be subtracted.
+        maskedModel : `~lsst.afw.image.MaskedImageF`
+            A masked image of the PSF model.
+
+        Returns
+        -------
+        annularFlux: float
+            The annular flux of the PSF model at the radius where the flux of the given star is determined.
+        """
+        annulusImage = self.initAnnulusImage()
+        annulus = self.createAnnulus(brightStarStamp)
+        annulus.copyMaskedImage(maskedModel, annulusImage)
+        annularFlux = self.applyStatsControl(annulusImage)
+        return annularFlux
+
+    def findPsfAnnularFluxes(self, brightStarStamps):
+        """Find the annular fluxes of the given PSF model.
+
+        Parameters
+        ----------
+        brightStarStamps : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamps`
+            The stamps of stars that will be subtracted from the exposure.
+
+        Returns
+        -------
+        PsfAnnularFluxes: numpy.array
+            A two column numpy.array containing annular fluxes of the PSF at radii where the flux for stars
+            exist (could be found).
+
+        Notes
+        -----
+        While the PSF model is normalized at a certain radius, the flux of a star at that radius might be
+        impossible to find. Therefore, we have to scale the PSF model considering a radius where the star has
```

Comment:
Do you mean the annular flux around that radius?

Reply:
Yep! The text is modified now!